drivers/ethernet/phy/phy_microchip_ksz8081.c

```diff
@@ -283,27 +276,27 @@ static int phy_mc_ksz8081_reset(const struct device *dev)
 	}
 
 #if DT_ANY_INST_HAS_PROP_STATUS_OKAY(reset_gpios)
-	if (!config->reset_gpio.port) {
-		goto skip_reset_gpio;
```

Comment:
the rest of the PR is good but I preferred the negative check here to avoid the extra indentation, maybe would be better to make the gpio reset into a helper function, like:

```
#if DT_ANY_INST_HAS_PROP_STATUS_OKAY
static int phy_ksz8081_reset_gpio(config *)
{
  int ret;

  if (!config->reset_gpio.port) {
    return 0;
  }

  /* rest of reset code */
  return ret;
}
#else 
#define phy_ksz8081_reset_gpio(...)
#endif

Reply:
Introducing helper functions might cause ambiguity between a missing GPIO port (returning 0) and a successful GPIO operation (also returning 0).
Adding extra return values to differentiate them would complicate the code.

Given the constraint of maintaining a max of 2 indents, I suggest reverting to the previous code structure but improving it by moving `const struct mc_ksz8081_config *config = dev->config;` inside the `#if` block.

WDYT? :)

Reply:
There's already silence if there's a missing gpios port, maybe what you want to do is add some kind of log message

Reply:
I kind of prefer the idea to move the #if outside of the function, with the helper function, I think this practice in general  makes the code more easy to work in and avoid starting to have build configuration complications